tortoise/algorithm.go

```diff
@@ -281,22 +353,25 @@ func (t *Tortoise) OnHareOutput(lid types.LayerID, bid types.BlockID) {
 	defer t.mu.Unlock()
 	waitHareOutputDuration.Observe(float64(time.Since(start).Nanoseconds()))
 	t.trtl.onHareOutput(lid, bid)
+	if t.tracer != nil {
+		t.tracer.On(&HareTrace{Layer: lid, Vote: bid})
+	}
 }
 
 // GetMissingActiveSet returns unknown atxs from the original list. It is done for a specific epoch
 // as active set atxs never cross epoch boundary.
-func (t *Tortoise) GetMissingActiveSet(epoch types.EpochID, atxs []types.ATXID) []types.ATXID {
+func (t *Tortoise) GetMissingActiveSet(epoch types.EpochID, atxs []types.ATXID) (missing []types.ATXID) {
 	t.mu.Lock()
 	defer t.mu.Unlock()
 	edata, exists := t.trtl.epochs[epoch]
 	if !exists {
-		return atxs
-	}
-	var missing []types.ATXID
-	for _, atx := range atxs {
-		_, exists := edata.atxs[atx]
-		if !exists {
-			missing = append(missing, atx)
+		missing = atxs
+	} else {
+		for _, atx := range atxs {
+			_, exists := edata.atxs[atx]
+			if !exists {
+				missing = append(missing, atx)
+			}
 		}
 	}
 	return missing
```

Comment:
I personally think the previous code was easier to understand than the changed code (functional they seem to be identical?)

Reply:
I agree, I don't see a reason for the else which creates additional nesting and makes it harder for me to read.

Reply:
the reason for this was that i added a trace for this method as well, but then decided to remove it temporarily.  with nesting i had to add trace in one place

Reply:
reverted